### PR TITLE
add basic git workflows - part 2

### PR DESCRIPTION
**Pull request by LeonieFreisinger**
🔬 Background

NeuralProphet has a convenient set of implemented git workflows. It is beneficial for test-of-time to implement a selection of these workflows as well.

🔮 Key changes
Add the following git workflows:
- linters: black, isort, flake8
- ci: pytests

📋 Review Checklist
- [x]  I have performed a self-review of my own code.
- [x]  I have commented my code, added docstrings and data types to function definitions.
- [x]  no pytests needed

### CODE REVIEW

**Comment by LeonieFreisinger** [comment]
@noxan would be great to get your support by reviewing the implemented git_workflows. P.S. I already double-checked the adapted files in the test and tot folder. They have many changes due to reformatting. You don´t have to double-check them again. Thanks a lot :)

File: pyproject.toml [hunk 1]
```diff
@@ -24,3 +24,10 @@ exclude = '''
   | .gitignore
 )
 '''
+
+[tool.isort]
+profile = "black"
+line_length = 120
+
+[tool.pyright]
+include = ["neuralprophet/configure.py"]
```
Comment (noxan) [inline pyproject.toml:33]: That looks like the wrong project, might make sense to start with proper typing for all files right from the start

File: requirements/dev.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+black[jupyter]
+flake8
+isort
+prophet
```
Comment (noxan) [inline requirements/dev.txt:4]: Do we need Prophet here? Was only added to NP due to some migration samples and tests

File: requirements/dev.txt [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+pyright
+pytest>=6.2.3
+pytest-cov
+sphinx>=4.2.0
+twine
+wheel
+
```
Comment (noxan) [inline requirements/dev.txt:11]: Duplicate newline

**Review by noxan — COMMENTED** [review]
Added some comments

**Pull request merged**